### PR TITLE
add parent team id when calling create team function fix black precom…

### PR DESCRIPTION
**Pull request by joosangkim**
1. create team with `parent team id`
  * Based on github [official document](https://docs.github.com/en/rest/teams/teams#create-a-team) , creating team with a parent team id is supported but `PyGithub` doesn't support it yet. So I add this feature on  existing function named `create_team`. 

2. solve `pre-commit` dependency error 
* when run `pre-commit`, there are dependency issue on https://github.com/psf/black@20.8b1. `psf/black` has import issue with `click`. You can check the issue on this [issue page](https://github.com/psf/black/issues/2976)

### CODE REVIEW

**Comment by mitchLui** [comment]
Is this still being worked on? I'd really appreciate it if this was available and can contribute if wanted.

**Comment by joosangkim** [comment]
Hi @mitchLui I’ve forgot this PR because there are so many `parent_team_id` related PR. I’ll resolving my PR about conflicts and failed test.

**Comment by Cheshirez** [comment]
Hi Is this PR still active?

**Comment by tobiasehlert** [comment]
Hi @joosangkim, are you still working on this PR?
Would be really nice to get this implemented :)

File: github/Organization.py [hunk 1]
```diff
@@ -1379,6 +1394,7 @@ def _initAttributes(self):
         self._type = github.GithubObject.NotSet
         self._updated_at = github.GithubObject.NotSet
         self._url = github.GithubObject.NotSet
+        self._parent_team_id = github.GithubObject.NotSet
 
     def _useAttributes(self, attributes):
```
Comment (EnricoMi) [inline github/Organization.py:1399]: Attribute `parent_team_id` of `Organization` has to be set in `_useAttributes`

File: tests/Organization.py [hunk 2]
```diff
@@ -160,6 +160,7 @@ def testCreateTeamWithAllArguments(self):
             "push",
             "secret",
             "Description also created by PyGithub",
+            141496,
```
Comment (EnricoMi) [inline tests/Organization.py:163]: Can we assert the `parent_team_id` in `testAttributes`? I may have to record replay data for this.

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Comment by EnricoMi** [comment]
> 2. solve `pre-commit` dependency error

@joosangkim Can we fix this in a separate PR? How come our CI does not break due to this error?

**Comment by EnricoMi** [comment]
@joosangkim happy to merge once comments are addressed

File: github/Organization.py (hunk 1 above)
Comment (EnricoMi) [reply github/Organization.py:1399]: Btw., I can't see in the docs that Organization has a `parent_team_id` attribute...

**Comment by EnricoMi** [comment]
Fixed by #2863.

**Pull request closed without merging**